src/presentation/api/controller/runtime.go

```diff
@@ -0,0 +1,166 @@
+package apiController
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/speedianet/os/src/domain/entity"
+	"github.com/speedianet/os/src/domain/valueObject"
+	internalDbInfra "github.com/speedianet/os/src/infra/internalDatabase"
+	apiHelper "github.com/speedianet/os/src/presentation/api/helper"
+	"github.com/speedianet/os/src/presentation/service"
+)
+
+type RuntimeController struct {
+	runtimeService *service.RuntimeService
+}
+
+func NewRuntimeController(
+	persistentDbService *internalDbInfra.PersistentDatabaseService,
+) *RuntimeController {
+	return &RuntimeController{
+		runtimeService: service.NewRuntimeService(persistentDbService),
+	}
+}
+
+// ReadPhpConfigs godoc
+// @Summary      ReadPhpConfigs
+// @Description  Get php version, modules and settings for a hostname.
+// @Tags         runtime
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Param        hostname 	  path   string  true  "Hostname"
+// @Success      200 {object} entity.PhpConfigs
+// @Router       /v1/runtime/php/{hostname}/ [get]
+func (controller *RuntimeController) ReadPhpConfigs(c echo.Context) error {
+	requestBody := map[string]interface{}{
+		"hostname": c.Param("hostname"),
+	}
+
+	return apiHelper.ServiceResponseWrapper(
+		c, controller.runtimeService.ReadPhpConfigs(requestBody),
+	)
+}
+
+func parsePhpModules(rawPhpModules interface{}) ([]entity.PhpModule, error) {
+	modules := []entity.PhpModule{}
+
+	rawModulesSlice, assertOk := rawPhpModules.([]interface{})
+	if !assertOk {
+		rawModuleUnit, assertOk := rawPhpModules.(map[string]interface{})
+		if !assertOk {
+			return modules, errors.New("InvalidPhpModules")
+		}
+		rawModulesSlice = []interface{}{rawModuleUnit}
+	}
+
+	for _, rawModule := range rawModulesSlice {
+		rawModuleMap, assertOk := rawModule.(map[string]interface{})
+		if !assertOk {
+			slog.Debug("PhpModuleIsNotAnInterface")
+			continue
+		}
+
+		moduleName, err := valueObject.NewPhpModuleName(rawModuleMap["name"])
+		if err != nil {
+			slog.Debug("InvalidPhpModuleName", slog.Any("name", rawModuleMap["name"]))
+			continue
+		}
+
+		moduleStatus, assertOk := rawModuleMap["status"].(bool)
```

Comment:
You should use the BoolParser, which by the way, I renamed to InterfaceToBool and put it on the voHelper area.

Reply:
Done!